mbl-core/tests/partitions/test_partitions.py

```diff
@@ -86,38 +86,39 @@ def get_device_file_for_parts():
 
     This function assumes that all of the partitions are on the same device.
     """
-    return subprocess.check_output(["blkid", "-L", "rootfs1"]).rsplit(b"p", 1)[
-        0
-    ]
+    return str(
+        subprocess.check_output(["blkid", "-L", "rootfs1"])
+        .rsplit(b"p", 1)[0]
+        .split(b"/")[2],
+        "utf-8",
+    )
 
 
 def get_actual_part_table():
     """
     Get the actual partition table.
 
     Returns a list of (offset, size) pairs for file system partitions, based on
-    the output of "sfdisk -J".
+    the content of /sys/block.
+
 
     The list is sorted and the offsets and sizes are in KiB units.
     """
     device_file = get_device_file_for_parts()
-    # "sfdisk -J" gives JSON output
-    sfdisk_json = subprocess.check_output(["sfdisk", "-J", device_file])
-
-    # The output contains some info we're not interested in - we just want the
-    # "partitions" list which contains dicts with an offsets and sizes for each
-    # partition.
-    sfdisk_part_table = json.loads(sfdisk_json)["partitiontable"]["partitions"]
-
-    # Convert the partition table from sfdisk into a list of (offset, size)
-    # tuples. Divide the numbers by 2 to convert 512B sectors that sfdisk uses
-    # into KiB
-    part_table = sorted(
-        [
-            (int(d["start"]) // 2, int(d["size"]) // 2)
-            for d in sfdisk_part_table
-        ]
-    )
+    device_dir = pathlib.Path.joinpath(ACTUAL_PART_INFO_DIR, device_file)
+
+    part_table = []
+
+    filename_re = re.compile(r"^.*{}p([0-9]+)$".format(device_file))
```

Comment:
What's the `.*` at the start of the regex for? You're just matching against filenames rather than full paths so there shouldn't be anything where the `.*` is, right?

Why are you capturing the `[0-9]+`?

I think you should `re.escape(device_file)`

```suggestion
    filename_re = re.compile(r"^{}p[0-9]+$".format(re.escape(device_file)))
```
Although it might be easier to just use a glob instead like
```Python
for f in device_dir.glob("{}p[0-9]*".format(glob.escape(device_file)):
```
That wouldn't match as strictly though.

Reply:
Changed to use glob.